### PR TITLE
began status endpoint and set up return json

### PR DESCRIPTION
**Pull request by aalvarez11**
created a controller for the status endpoint and set up the json with the service and timestamp but still working on getting the play environment mode to display as a name rather than a string of letters and numbers

### CODE REVIEW

File: app/controllers/APIController.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import play.api.libs.json._
+import play.api.Environment._
+
+import java.time.format.DateTimeFormatter
+
+/**
+ * This controller creates an `Action` to handle HTTP requests to the
+ * application's home page.
+ */
+@Singleton
```
Comment (fsasieta) [inline app/controllers/APIController.scala:16]: This comment is non descriptive, please remove it

File: conf/routes [hunk 2]
```diff
@@ -9,3 +9,6 @@ GET     /hello                      controllers.HomeController.hello()
 
 # Map static resources from the /public folder to the /assets URL path
 GET     /assets/*file               controllers.Assets.versioned(path="/public", file: Asset)
+
+# API endpoints
+GET     /pass-bakery/status         controllers.APIController.getStatus()
```
Comment (fsasieta) [inline conf/routes]: You're missing and end of line character in this file, please add it

File: app/views/hello.scala.html [hunk 3]
```diff
@@ -5,7 +5,7 @@
                 <h1>Hello World</h1>
             </article>
             <aside>
-                @commonSidebar()
+
```
Comment (fsasieta) [inline app/views/hello.scala.html:8]: Was this function not defined before?

File: app/controllers/APIController.scala [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    
+    play.api.Mode.Prod
+
+    // construct the json
+    val json: JsValue = Json.obj(
+      "service" -> "pass-bakery",
+      "environment" -> play.api.Mode.toString,
+      "test" -> play.api.Mode.Dev.toString,
+      "serverTime" -> LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"))
+    )
```
Comment (fsasieta) [inline app/controllers/APIController.scala]: Good start! A thing to keep in mind is that it is always better to define the models you return than build JSON or other serialized formats ad hoc. In this case, I would suggest defining a [case class](https://docs.scala-lang.org/tour/case-classes.html) in file in the `app/models` directory that describes what you will send to the the user/api consumer in this case, and instantiate the case class here with any values that need to be defined or calculated when the response is returned to the user

File: app/controllers/APIController.scala [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+@Singleton
+class APIController @Inject()(val controllerComponents: ControllerComponents) extends BaseController {
+
+  /*
+   * method that returns a json object with the following information:
+   * { "service" : "pass-bakery"
+   *   "environment" : string (production or development run, based on play fw env config)
+   *   "serverTime" : string (local time from dateTime library in ISO 8601 format)
+   * */
+  def getStatus() = Action { implicit request: Request[AnyContent] =>
```
Comment (fsasieta) [inline app/controllers/APIController.scala:26]: Generally, it is preferable to declare comments in scala docs (but this is not a rule enforced everywhere). Those comments would have the inner format `/** */` (one extra asterisk) and they generally appear more visible in my IntelliJ installation.

File: app/controllers/APIController.scala [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    
+    play.api.Mode.Prod
+
+    // construct the json
```
Comment (fsasieta) [inline app/controllers/APIController.scala]: Comments should tell you the why, not the how something is done. I'm able to read that you are declaring a json here, but don't have any insight as to why you are doing that

**Review by fsasieta — CHANGES_REQUESTED** [review]

File: app/controllers/APIController.scala (hunk 1 above)
Comment (aalvarez11) [reply app/controllers/APIController.scala:16]: comment was removed

File: app/controllers/APIController.scala (hunk 5 above)
Comment (aalvarez11) [reply app/controllers/APIController.scala:26]: added the second asterisk to show the comment block as a scala doc

File: app/controllers/APIController.scala (hunk 6 above)
Comment (aalvarez11) [reply app/controllers/APIController.scala]: removed the comment

File: app/controllers/APIController.scala (hunk 4 above)
Comment (aalvarez11) [reply app/controllers/APIController.scala]: added the case class model under a new models directory and implemented it in the API controller

File: app/views/hello.scala.html (hunk 3 above)
Comment (aalvarez11) [reply app/views/hello.scala.html:8]: I deleted it for the sake of the program going forward

File: conf/routes (hunk 2 above)
Comment (aalvarez11) [reply conf/routes]: added a new line to routes as suggested

**Review by aalvarez11 — COMMENTED** [review]
addressed all the comments, awaiting instruction

File: app/controllers/APIController.scala [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  /**
+   * method that returns a json object with the following information:
+   * { "service" : "pass-bakery"
+   *   "environment" : string (production or development run, based on play fw env config)
+   *   "serverTime" : string (local time from dateTime library in ISO 8601 format)
+   * }
+   */
+
+  def getStatus() = Action { implicit request: Request[AnyContent] =>
+    implicit val statusWrites: OWrites[StatusInfo] = Json.writes[StatusInfo]
```
Comment (fsasieta) [inline app/controllers/APIController.scala]: Read about [companion objects](https://docs.scala-lang.org/overviews/scala-book/companion-objects.html) in scala. You should be able to move this line inside an object of the name `StatusInfo` that lives inside the `StatusInfo` file.

File: app/controllers/APIController.scala [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    val userEnvironment = env.mode.toString
+    val formattedTimestamp = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"))
+    val userInfo = StatusInfo("pass-bakery", userEnvironment, formattedTimestamp)
+    val statusJson : JsValue = Json.toJson(userInfo)
```
Comment (fsasieta) [inline app/controllers/APIController.scala]: It's usually best practice to extract any logic not related to HTTP results from the controller. Can you make a function that instantiates the object needed and call the function here?

**Review by fsasieta — COMMENTED** [review]
Can you post instructions on how to get the different values of env when running the application? (how did you verify functionality?)

File: app/controllers/APIController.scala (hunk 7 above)
Comment (aalvarez11) [reply app/controllers/APIController.scala]: The implicit line has been moved to a companion object StatusInfo inside the case class StatusInfo

**Comment by aalvarez11** [comment]
In order to display what mode the play framework environment is currently running on, the environment must be injected into the service or controller that handles the logic. 

1. After the class declaration, add the Inject annotation and after the annotation's parentheses, add another set of parentheses with an environment parameter and local variable name of choice, here 'env' is my local variable
`@Inject()(env: Environment)`
2. In the logic, the local environment variable you declared can be used and has a member 'mode' that will read either Dev, Prod, or Test, depending on the mode the play app is run in. The mode is an enum so it must be converted to a string, but the member has a toString method to accomplish this.
`env.mode.toString`
3. To test what environment is being fetched, go to the terminal. The command I used in the terminal to run the application in development mode is `sbt run` and to run it in production mode, I used `sbt runProd`. However, be aware that running the application in production mode will continue to run and use the default 9000 port. To terminate production mode, use the mac terminal command `lsof -i tcp:9000` to see the process id (PID) of processes using port 9000, and kill them with terminal command `kill <PID>` for any (if more than one) process(es) running that are using port 9000. 

The results for my application's use of environment yield the following results when run in Dev (Developer) mode and Prod (Production) mode respectively:

Screen Shot 2022-08-05 at 11 12 34 AM

Screen Shot 2022-08-05 at 11 10 26 AM

**Review by fsasieta — APPROVED** [review]

**Pull request merged**